procedures/unit-testing-tracing-analytics.ipf

```diff
@@ -6,4 +6,30 @@
 #if (IgorVersion() >= 9.00) && Exists("TUFXOP_Version") && (NumberByKey("BUILD", IgorInfo(0)) >= 38812)
 
 
+static Function/WAVE GetTotals()
+	variable i, numWaves
+
+	TUFXOP_GetStorage/N="IUTF_TestRun" storage
+	numWaves = DimSize(storage, UTF_ROW)
+	WAVE/Z totals
```

Comment:
I guess you want a null wave ref here.
Thus, it would be `WAVE/ZZ totals` (or oldschool `WAVE/Z totals = $""`)
The problem with `WAVE/Z totals` is that it trys to resolve a wave named `totals` in the current datafolder, because it is short for `WAVE/Z totals = totals`.
So there is a chance that you get a wave reference to a unrelated wave and you continue corrupting this data.

Though, from the logic here, at the end you check if you actually created totals and if not create it with zero size.
You could wrap that around and initially create a wave with zero size with `Make/FREE=1/N=0 totals` and change the WaveExists check to a zero size check?

Reply:
With

```igorpro
Function CreateWaves()

	Make/O wv1
	Make/O wv2
	Make/O wv3
End

Function Test()

	WAVE/Z wv1
	print WaveExists(wv1)

	WAVE/Z wv2 = $""
	print WaveExists(wv2)

	WAVE/ZZ wv3
	print WaveExists(wv3)
End
```

I get 

```
•test()
  1
  0
  0
```

so Michael is right that just /Z doesn't cut it. I'd use `Wave/ZZ totals`.